### ISSUE TITLE
Error with WMTS example

### ISSUE THREAD

**Issue by dopplershift** [geopython/OWSLib#619]
The code example from the docs:
```python
>>> from owslib.wmts import WebMapTileService
>>> wmts = WebMapTileService("http://map1c.vis.earthdata.nasa.gov/wmts-geo/wmts.cgi")
```
currently results in a traceback:
```pytb
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-d6e4fa9c2cd1> in <module>
      4 fig = plt.figure()
      5 ax = fig.add_subplot(1, 1, 1, projection=ccrs.PlateCarree())
----> 6 ax.add_wmts(url, layer)
      7 ax.set_extent([-15, 25, 35, 60], crs=ccrs.PlateCarree())
      8 

~/miniconda3/envs/py37/lib/python3.7/site-packages/cartopy/mpl/geoaxes.py in add_wmts(self, wmts, layer_name, wmts_kwargs, **kwargs)
   1939         from cartopy.io.ogc_clients import WMTSRasterSource
   1940         wmts = WMTSRasterSource(wmts, layer_name,
-> 1941                                 gettile_extra_kwargs=wmts_kwargs)
   1942         return self.add_raster(wmts, **kwargs)
   1943 

~/miniconda3/envs/py37/lib/python3.7/site-packages/cartopy/io/ogc_clients.py in __init__(self, wmts, layer_name, gettile_extra_kwargs)
    378                 hasattr(wmts, 'contents') and
    379                 hasattr(wmts, 'gettile')):
--> 380             wmts = owslib.wmts.WebMapTileService(wmts)
    381 
    382         try:

~/miniconda3/envs/py37/lib/python3.7/site-packages/owslib/wmts.py in __init__(self, url, version, xml, username, password, parse_remote_metadata, vendor_kwargs, auth)
    189 
    190         # build metadata objects
--> 191         self._buildMetadata(parse_remote_metadata)
    192 
    193     def _getcapproperty(self):

~/miniconda3/envs/py37/lib/python3.7/site-packages/owslib/wmts.py in _buildMetadata(self, parse_remote_metadata)
    239                     self.contents[cm.id] = cm
    240                 gather_layers(elem, cm)
--> 241         gather_layers(caps, None)
    242 
    243         self.tilematrixsets = {}

~/miniconda3/envs/py37/lib/python3.7/site-packages/owslib/wmts.py in gather_layers(parent_elem, parent_metadata)
    232                 cm = ContentMetadata(
    233                     elem, parent=parent_metadata, index=index + 1,
--> 234                     parse_remote_metadata=parse_remote_metadata)
    235                 if cm.id:
    236                     if cm.id in self.contents:

~/miniconda3/envs/py37/lib/python3.7/site-packages/owslib/wmts.py in __init__(self, elem, parent, index, parse_remote_metadata)
    716 
    717         link_elements = elem.findall(_TILE_MATRIX_SET_LINK_TAG)
--> 718         tile_matrix_set_links = TileMatrixSetLink.from_elements(link_elements)
    719         self.tilematrixsetlinks = {}
    720         for tmsl in tile_matrix_set_links:

~/miniconda3/envs/py37/lib/python3.7/site-packages/owslib/wmts.py in from_elements(link_elements)
    655                             msg = ('TileMatrixLimits with tileMatrix "%s" '
    656                                    'already exists' % tml.tilematrix)
--> 657                             raise KeyError(msg)
    658                         tilematrixlimits[tml.tilematrix] = tml
    659 

KeyError: 'TileMatrixLimits with tileMatrix "1" already exists'
```

This is with owslib 0.18.0 (on Mac, but I doubt that's relevant). No idea what's going on or who's to blame here.

**Comment by cehbrecht** [comment]
I can't say if it is the service:
http://map1c.vis.earthdata.nasa.gov/wmts-geo/wmts.cgi

or the module.

Is there another service we can use for testing?

**Comment by cehbrecht** [comment]
These failing WMTS tests are skipped by PR #620.

**Comment by QuLogic** [comment]
The `TileMatrixSetLimits` from this service returns multiple `TileMatrixLimits` with the same `TileMatrix`:
```
(Pdb) print(etree.tostring(link_element))
```
```xml
<ns0:TileMatrixSetLink xmlns:ns0="http://www.opengis.net/wmts/1.0">
  <ns0:TileMatrixSet>2km</ns0:TileMatrixSet>
  <ns0:TileMatrixSetLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>0</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>0</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>1</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>1</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>1</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>0</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>1</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>1</ns0:MaxTileRow>
      <ns0:MinTileCol>2</ns0:MinTileCol>
      <ns0:MaxTileCol>2</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>2</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>2</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>1</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>2</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>2</ns0:MaxTileRow>
      <ns0:MinTileCol>4</ns0:MinTileCol>
      <ns0:MaxTileCol>4</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>3</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>4</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>3</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>3</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>4</ns0:MaxTileRow>
      <ns0:MinTileCol>8</ns0:MinTileCol>
      <ns0:MaxTileCol>9</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>4</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>9</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>6</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>4</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>9</ns0:MaxTileRow>
      <ns0:MinTileCol>17</ns0:MinTileCol>
      <ns0:MaxTileCol>19</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>5</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>18</ns0:MaxTileRow>
      <ns0:MinTileCol>0</ns0:MinTileCol>
      <ns0:MaxTileCol>13</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
    <ns0:TileMatrixLimits>
      <ns0:TileMatrix>5</ns0:TileMatrix>
      <ns0:MinTileRow>0</ns0:MinTileRow>
      <ns0:MaxTileRow>18</ns0:MaxTileRow>
      <ns0:MinTileCol>35</ns0:MinTileCol>
      <ns0:MaxTileCol>39</ns0:MaxTileCol>
    </ns0:TileMatrixLimits>
  </ns0:TileMatrixSetLimits>
</ns0:TileMatrixSetLink>
```
But is that a bug with the service? It's not clear to me from the spec that `TileMatrix` should be unique.

**Comment by raphaeldussin** [comment]
FYI all the examples out there using cartopy and wmts are broken (e.g. https://scitools.org.uk/cartopy/docs/v0.15/examples/wmts.html)

the problem can be traced back to:

https://github.com/geopython/OWSLib/blob/5f9751c7b1c9108b967270dfbb7a41bf7068999a/owslib/wmts.py#L642

I haven't dug into the code to understand what the logic is for that error message
but basically commenting it out fixed all the examples I have tried so far.

I'm happy to send a PR with error -> warning

I'm sure a lot of people would be happy to have their cartopy/wmts code working again.

**Comment by cehbrecht** [comment]
@raphaeldussin Thanks for taking care of this. A PR is welcome :)

**Comment by raphaeldussin** [comment]
PR is #656

**Comment by cehbrecht** [comment]
closed by PR #656.
